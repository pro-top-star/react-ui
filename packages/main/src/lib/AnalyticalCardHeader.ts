import { AnalyticalCardHeader } from '../components/AnalyticalCard/header/AnalyticalCardHeader';

export { AnalyticalCardHeader };
