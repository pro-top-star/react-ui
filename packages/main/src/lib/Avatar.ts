import { Avatar } from '../webComponents/Avatar';
import type { AvatarPropTypes } from '../webComponents/Avatar';

export { Avatar };
export type { AvatarPropTypes };
