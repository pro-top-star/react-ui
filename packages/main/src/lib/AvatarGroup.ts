import { AvatarGroup } from '../webComponents/AvatarGroup';
import type { AvatarGroupPropTypes } from '../webComponents/AvatarGroup';

export { AvatarGroup };
export type { AvatarGroupPropTypes };
