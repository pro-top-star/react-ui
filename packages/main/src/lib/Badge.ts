import { Badge } from '../webComponents/Badge';
import type { BadgePropTypes } from '../webComponents/Badge';

export { Badge };
export type { BadgePropTypes };
