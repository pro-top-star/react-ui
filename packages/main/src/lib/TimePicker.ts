import { TimePicker } from '../webComponents/TimePicker';
import type { TimePickerPropTypes } from '../webComponents/TimePicker';

export { TimePicker };
export type { TimePickerPropTypes };
