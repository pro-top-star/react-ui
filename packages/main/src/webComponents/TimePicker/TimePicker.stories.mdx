import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';
import { TimePicker } from '@ui5/webcomponents-react/lib/TimePicker';
import { ValueState } from '@ui5/webcomponents-react/lib/ValueState';
import { CSSProperties, Ref } from 'react';
import { createSelectArgTypes } from '@shared/stories/createSelectArgTypes';
import { DocsHeader } from '@shared/stories/DocsHeader';
import { DocsCommonProps } from '@shared/stories/DocsCommonProps';

<Meta
  title="UI5 Web Components / TimePicker"
  component={TimePicker}
  argTypes={{
    ...createSelectArgTypes({ valueState: ValueState }),
    valueStateMessage: { control: { disable: true } },
    ...DocsCommonProps
  }}
  args={{
    valueState: ValueState.None,
    style: {},
    className: '',
    tooltip: '',
    slot: '',
    ref: null
  }}
/>

<DocsHeader />

<Canvas>
  <Story name="Default">
    {(args) => {
      return <TimePicker {...args} />;
    }}
  </Story>
</Canvas>

<ArgsTable story="." />
<div style={{ fontFamily: 'var(--sapFontFamily)', fontSize: 'var(--sapFontSize)', color: 'var(--sapTextColor)' }}>
  <h3>Usage</h3>
  The user can enter a time by:
  <ul>
    <li>Using the sliders that opens in a popup</li>
    <li>Typing it in directly in the input field</li>
  </ul>
  <br />
  <br />
  When the user makes an entry and chooses the enter key, the sliders shows the corresponding time. When the user directly
  triggers the sliders display, the actual time is displayed.
</div>
