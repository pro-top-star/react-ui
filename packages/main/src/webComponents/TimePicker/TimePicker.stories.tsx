import { createSelectArgTypes } from '@shared/stories/createSelectArgTypes';
import { TimePicker } from '@ui5/webcomponents-react/lib/TimePicker';
import { ValueState } from '@ui5/webcomponents-react/lib/ValueState';
import React from 'react';

export default {
  title: 'UI5 Web Components / TimePicker',
  component: TimePicker,
  argTypes: {
    ...createSelectArgTypes({ valueState: ValueState }),
    ref: {
      type: null
    }
  },
  args: {
    formatPattern: 'hh:mm:ss a',
    valueState: ValueState.None
  }
};

export const generatedDefaultStory = (props) => (
  <TimePicker
    disabled={props.disabled}
    formatPattern={props.formatPattern}
    placeholder={props.placeholder}
    readonly={props.readonly}
    value={props.value}
    valueState={props.valueState}
    onChange={props.onChange}
    onInput={props.onInput}
  />
);

generatedDefaultStory.storyName = 'Generated default story';
