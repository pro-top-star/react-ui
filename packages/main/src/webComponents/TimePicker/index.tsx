import { ValueState } from '@ui5/webcomponents-react/lib/ValueState';
import { withWebComponent, WithWebComponentPropTypes } from '@ui5/webcomponents-react/lib/withWebComponent';
import '@ui5/webcomponents/dist/TimePicker';
import { FC, ReactNode } from 'react';

export interface TimePickerPropTypes extends Omit<WithWebComponentPropTypes, 'onChange' | 'onInput'> {
  /**
   * Determines whether the <code>ui5-time-picker</code> is displayed as disabled.
   */
  disabled?: boolean;
  /**
   * Determines the format, displayed in the input field.<br/><br/>
   *
   * Example: HH:mm:ss -> 11:42:35 hh:mm:ss a -> 2:23:15 PM mm:ss -> 12:04 (only minutes and seconds)
   */
  formatPattern?: string;
  /**
   * Defines a short hint, intended to aid the user with data entry when the <code>ui5-time-picker</code> has no value.<br/><br/>
   *
   * <b>Note:</b> When no placeholder is set, the format pattern is displayed as a placeholder. Passing an empty string as the value of this property will make the <code>ui5-time-picker</code> appear empty - without placeholder or format pattern.
   */
  placeholder?: string;
  /**
   * Determines whether the <code>ui5-time-picker</code> is displayed as readonly.
   */
  readonly?: boolean;
  /**
   * Defines a formatted time value.
   */
  value?: string;
  /**
   * Defines the value state of the <code>ui5-time-picker</code>. <br><br> Available options are: <ul> <li><code>None</code></li> <li><code>Error</code></li> <li><code>Warning</code></li> <li><code>Success</code></li> <li><code>Information</code></li> </ul>
   */
  valueState?: ValueState;
  /**
   * Defines the value state message that will be displayed as pop up under the <code>ui5-time-picker</code>. <br><br><br/><br/>
   *
   * <b>Note:</b> If not specified, a default text (in the respective language) will be displayed. <br> <b>Note:</b> The <code>valueStateMessage</code> would be displayed, when the <code>ui5-time-picker</code> is in <code>Information</code>, <code>Warning</code> or <code>Error</code> value state.
   */
  valueStateMessage?: ReactNode;
  /**
   * Fired when the input operation has finished by clicking the "OK" button or when the text in the input field has changed and the focus leaves the input field.
   */
  onChange?: (event: CustomEvent<{}>) => void;
  /**
   * Fired when the value of the <code>ui5-time-picker</code> is changed at each key stroke.
   */
  onInput?: (event: CustomEvent<{}>) => void;
}

/**
 * <code>import { TimePicker } from '@ui5/webcomponents-react/lib/TimePicker';</code>
 * <br />
 * <a href="https://sap.github.io/ui5-webcomponents/playground/components/TimePicker" target="_blank">UI5 Web Components Playground</a>
 */
const TimePicker: FC<TimePickerPropTypes> = withWebComponent<TimePickerPropTypes>(
  'ui5-time-picker',
  ['formatPattern', 'placeholder', 'value', 'valueState'],
  ['disabled', 'readonly'],
  ['valueStateMessage'],
  ['change', 'input']
);

TimePicker.displayName = 'TimePicker';

TimePicker.defaultProps = {
  disabled: false,
  placeholder: undefined,
  readonly: false,
  value: undefined,
  valueState: ValueState.None
};

export { TimePicker };
