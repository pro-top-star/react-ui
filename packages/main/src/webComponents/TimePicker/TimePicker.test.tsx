import { TimePicker } from '@ui5/webcomponents-react/lib/TimePicker';
import { mount } from 'enzyme';
import React from 'react';

describe('TimePicker', () => {
  test('Basic Test (generated)', () => {
    const wrapper = mount(<TimePicker />);
    expect(wrapper.render()).toMatchSnapshot();
  });
});
