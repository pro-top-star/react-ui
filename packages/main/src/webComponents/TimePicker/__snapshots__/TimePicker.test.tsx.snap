// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimePicker Basic Test (generated) 1`] = `
<ui5-time-picker
  value-state="None"
/>
`;
